types/Insurance/Insurance.d.ts

```diff
@@ -0,0 +1,142 @@
+   * An identifying number for some insurance providers used by EasyPost
+   */
+  provider_id: string;
+
+  /**
+   * The ID of the Shipment in EasyPost, if postage was purchased via EasyPost
+   */
+  shipment_id: string;
+
+  /**
+   * The tracking code of either the shipment within EasyPost, or provided by you during creation
+   */
+  tracking_code: string;
+
+  /**
+   * The current status of the insurance, possible values are "new", "pending", "purchased", "failed", or "cancelled"
+   */
+  status: TInsuranceStatus;
+
+  /**
+   * The associated Tracker object
+   */
+  tracker: ITracker;
+
+  /**
+   * The associated Address object for destination
+   */
+  to_address: IAddress;
+
+  /**
+   * The associated Address object for origin
+   */
+  from_address: IAddress;
+
+  /**
+   * The associated InsuranceFee object if any
+   */
+  fee: IFee;
+
+  /**
+   * The list of errors encountered during attempted purchase of the insurance
+   */
+  messages: string[];
+}
+
+export declare class Insurance implements IInsurance {
+  public constructor(input: IInsuranceCreateParameters);
+
+  id: string;
+  mode: 'test' | 'production';
+  object: 'Insurance';
+  reference?: string | null;
+  amount: string;
+  provider: string;
+  provider_id: string;
+  shipment_id: string;
+  tracking_code: string;
+  status: TInsuranceStatus;
+  tracker: ITracker;
+  to_address: IAddress;
+  from_address: IAddress;
+  fee: IFee;
+  messages: string[];
+  created_at: string;
+  updated_at: string;
+
+  /**
+   * An Insurance created via this endpoint must belong to a shipment purchased outside of EasyPost.
+   * Insurance for Shipments created within EasyPost must be created via the Shipment Buy or Insure endpoints.
+   * When creating Insurance for a non-EasyPost shipment, you must provide to_address, from_address, tracking_code, and amount information.
+   * Optionally, you can provide the carrier parameter, which will help EasyPost identify the carrier the package was shipped with.
+   * If no carrier is provided, EasyPost will attempt to determine the carrier based on the tracking_code provided.
+   * Providing a carrier parameter is recommended, since some tracking_codes are ambiguous and may match with more than one carrier.
+   * In addition, not having to auto-match the carrier will significantly speed up the response time.
+   *
+   * @see https://www.easypost.com/docs/api/node#create-an-insurance
+   */
+  public save(): Promise<Insurance>;
+
+  /**
+   * The Insurance List is a paginated list of all Insurance objects associated with the given API Key.
+   * It accepts a variety of parameters which can be used to modify the scope.
+   * The has_more attribute indicates whether or not additional pages can be requested.
+   * The recommended way of paginating is to use either the before_id or after_id parameter to specify where the next page begins.
+   *
+   * @see https://www.easypost.com/docs/api/node#retrieve-a-list-of-insurances
+   */
+  static all(
```

Comment:
Maybe I was confused on earlier comments thinking that we forgot the `public` keyword. Are all static methods inherently public to begin with?

Reply:
yes, all methods. properties, static or member, are public by default.